packages/manager/modules/log-to-customer/src/service.js

```diff
@@ -1,8 +1,73 @@
 export default class LogToCustomerService {
   /* @ngInject */
-  constructor($http, $q) {
+  constructor($http, $q, iceberg, Poller) {
     this.$http = $http;
     this.$q = $q;
+    this.iceberg = iceberg;
+    this.Poller = Poller;
+  }
+
+  getRetention(serviceName, clusterId, retentionId) {
+    return this.$http
+      .get(
+        `/dbaas/logs/${serviceName}/cluster/${clusterId}/retention/${retentionId}`,
+      )
+      .then(({ data }) => data);
+  }
+
+  icebergQuery(url) {
+    return this.iceberg(url)
+      .query()
+      .expand('CachedObjectList-Cursor')
+      .execute(null, true)
+      .$promise.then(({ data }) => data);
+  }
+
+  get(url) {
+    return this.$http.get(url).then((data) => data);
+  }
+
+  post(url, payload) {
+    return this.$http.post(url, payload).then((data) => data);
```

Comment:
If you don't wrap a sub object in the response, `then` callback is useless
```suggestion
    return this.$http.post(url, payload);
```